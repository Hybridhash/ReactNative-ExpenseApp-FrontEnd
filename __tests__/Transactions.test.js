
import React from 'react';
import renderer from 'react-test-renderer';
import Transaction from '../src/components/Transactions';

jest.mock('@react-native-async-storage/async-storage', () => require('@react-native-async-storage/async-storage/jest/async-storage-mock'));

describe('Transaction display  Component', () => {
    it('renders component correctly', () => {
        const tree = renderer.create(
            <Transaction
              title="Test title"
              amount="Test amount"
              formattedDate="Test date"
              rightSwipeActions={() => {}}
            />
          ).toJSON();
          expect(tree).toMatchSnapshot();
    });
});
