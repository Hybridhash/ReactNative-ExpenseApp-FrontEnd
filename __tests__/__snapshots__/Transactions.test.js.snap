// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transaction display  Component renders component correctly 1`] = `
<View
  collapsable={false}
  forwardedRef={[Function]}
  handlerTag={1}
  handlerType="PanGestureHandler"
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onLayout={[Function]}
  style={
    {
      "overflow": "hidden",
    }
  }
>
  <View
    collapsable={false}
    style={
      {
        "bottom": 0,
        "flexDirection": "row-reverse",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "transform": [
          {
            "translateX": -10000,
          },
        ],
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#ff8303",
          "borderRadius": 10,
          "justifyContent": "center",
          "marginVertical": 6,
          "opacity": 1,
          "paddingHorizontal": 30,
        }
      }
    >
      <Text
        style={
          {
            "color": "#1b1a17",
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      >
        Delete
      </Text>
    </View>
    <View
      onLayout={[Function]}
    />
  </View>
  <View
    collapsable={false}
    forwardedRef={[Function]}
    handlerTag={2}
    handlerType="TapGestureHandler"
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    pointerEvents="auto"
    style={
      {
        "transform": [
          {
            "translateX": -0,
          },
        ],
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#4455BB",
          "borderColor": "#4455BB",
          "borderRadius": 10,
          "borderWidth": 3,
          "flexDirection": "column",
          "justifyContent": "space-between",
          "marginHorizontal": 16,
          "marginVertical": 8,
          "padding": 10,
        }
      }
    >
      <View
        style={
          {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 20,
              "fontWeight": "bold",
            }
          }
        >
          Test title
        </Text>
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 20,
              "fontWeight": "bold",
            }
          }
        >
          Test amount
        </Text>
      </View>
      <Text
        style={
          {
            "alignSelf": "center",
            "color": "#D3D3D3",
            "fontSize": 14,
            "fontWeight": "bold",
            "marginTop": 5,
          }
        }
      >
         Transaction date: 
        Invalid Date
      </Text>
    </View>
  </View>
</View>
`;
